src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -83,16 +83,19 @@ mod imp {
     // were originally supposed to do.
     //
     // This handler currently exists purely to print an informative message
-    // whenever a thread overflows its stack. When run the handler always
-    // un-registers itself after running and then returns (to allow the original
-    // signal to be delivered again). By returning we're ensuring that segfaults
-    // do indeed look like segfaults.
+    // whenever a thread overflows its stack. We then abort to exit with an
+    // error (and dump core if necessary), but to avoid a misleading SIGSEGV
```

Comment:
NIT: The 2nd part of this sentence has no weight behind it whatsoever, because SIGSEGV is accompanied with a overflow message, thus resolving any potential confusion that might arise.